cmd/amp-cluster/bootstrap.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+	"log"
+)
+
+var bootstrapCmd = &cobra.Command{
+	Use:     "bootstrap",
+	Short:   "Bootstraps the node",
+	Long:    `Installs Docker, setups Swarm, run the AMP agent`,
+	Aliases: []string{"bs"},
+	Run: func(cmd *cobra.Command, args []string) {
+		client.bootstrap(cmd, args)
+	},
+}
+
+// Flag variables
+var SwarmManagerHost string
+var SwarmManagerPort string
+var SwarmJoinToken string
+var AmpImageTag string
+var CreateSwarm bool
+var Manager bool
+
+func init() {
+	bootstrapCmd.Flags().BoolVarP(&Manager, "manager", "m", false, "bootstraps a Swarm manager node")
+	bootstrapCmd.Flags().BoolVarP(&CreateSwarm, "create", "c", false, "initializes the Swarm (needs option -m)")
+	bootstrapCmd.Flags().StringVarP(&SwarmJoinToken, "token", "", "", "Swarm join token")
+	bootstrapCmd.Flags().StringVarP(&SwarmManagerHost, "host", "", "", "Swarm manager hostname")
+	bootstrapCmd.Flags().StringVarP(&SwarmManagerPort, "port", "", "2377", "Swarm manager port")
+	bootstrapCmd.Flags().StringVarP(&AmpImageTag, "tag", "", Version, "AMP image version")
+	RootCmd.AddCommand(bootstrapCmd)
+}
+
+// Docker daemon install/update with the official Docker script
+// update the Docker configuration with:
+// - ulimit for file descriptors
+// - remote API
+// enable and start the Docker daemon
+// join a Swarm manager (or init the swarm)
+// if on a manager, create:
+// overlay networks amp-public and amp-infra,
+// a global service with the amp cluster server
+// a global service with the amp cluster agent
+func (c *clusterClient) bootstrap(cmd *cobra.Command, args []string) {
+	log.Println("Bootstrapping the node with CLI version ", Version)
+
+	var version string
+
+	_ = c.startDockerEngine()
+	version, _ = c.getDockerServerVersion()
+	ok, err := c.validateInstalledDockerEngineVersion(version)
+	if err != nil {
+		log.Fatalf("Failed to validate Docker version: %s\n", err)
+	}
+	if !ok {
+		if version != "0.0.0" {
+			log.Printf("Installed version %s is lower than %s\n", version, minimumDockerEngineVersion)
+		}
+		if err := c.installDockerEngine(); err != nil {
+			log.Fatalf("Failed to install Docker engine: %s\n", err)
+		}
+		_ = c.startDockerEngine()
+		version, err := c.getDockerServerVersion()
+		if err != nil {
+			log.Fatalf("Failed to get Docker version: %s\n", err)
+		}
+		ok, err = c.validateInstalledDockerEngineVersion(version)
+		if err != nil {
+			log.Fatalf("Failed to validate Docker version: %s\n", err)
+		}
+		if !ok {
+			log.Fatalf("Docker version does not meet the requirements: %s\n", version)
+		}
+	} else {
+		log.Println("Docker engine version is just fine")
+	}
+	if err := c.enableDockerEngine(); err != nil {
+		log.Fatalf("Failed to enable the Docker service: %s\n", err)
```

Comment:
use g.fatalc("%v\n", err) in place of log.Fatalln to have color output
globally use g.printfc(...) in place of log to have color output the same way then in the orther commands
including in bootstrap-docker.go file